ruby/ql/lib/codeql/ruby/frameworks/GraphQL.qll

```diff
@@ -0,0 +1,270 @@
+    this.getScopeExpr().(ConstantAccess).getName() = "Schema" and
+    this.getScopeExpr().(ConstantAccess).getScopeExpr().(ConstantAccess).getName() = "GraphQL"
+  }
+}
+
+private class GraphqlSchemaResolverAccess extends ConstantReadAccess {
+  //GraphQL::Schema::Resolver
+  GraphqlSchemaResolverAccess() {
+    this.getName() = "Resolver" and
+    this.getScopeExpr().(ConstantAccess).getName() = "Schema" and
+    this.getScopeExpr().(ConstantAccess).getScopeExpr().(ConstantAccess).getName() = "GraphQL"
+  }
+}
+
+private class GraphqlSchemaObjectAccess extends ConstantReadAccess {
+  //GraphQL::Schema::Object
+  GraphqlSchemaObjectAccess() {
+    this.getName() = "Object" and
+    this.getScopeExpr().(ConstantAccess).getName() = "Schema" and
+    this.getScopeExpr().(ConstantAccess).getScopeExpr().(ConstantAccess).getName() = "GraphQL"
+  }
+}
+
+
+/**
+ * A `ClassDeclaration` for a class that extends `GraphQL::Schema::RelayClassicMutation`.
+ * For example,
+ *
+ * ```rb
+ * module Mutations
+ *   class BaseMutation < GraphQL::Schema::RelayClassicMutation
+ *     argument_class Types::BaseArgument
+ *     field_class Types::BaseField
+ *     input_object_class Types::BaseInputObject
+ *     object_class Types::BaseObject
+ *   end
+ * end
+ *
+ * module Mutations
+ *   class MyMutation < BaseMutation
+ *     argument :something_id, ID, required: false
+ *     field :success, Boolean, null: false
+ *
+ *     def resolve(something_id:)
+ *       # call your application logic here...
+ *     end
+ *   end
+ * end
+ * ```
+ */
+class GraphqlRelayClassicMutationClass extends ClassDeclaration {
+  GraphqlRelayClassicMutationClass() {
+    // class BaseMutation < GraphQL::Schema::RelayClassicMutation
+    this.getSuperclassExpr() instanceof GraphqlRelayClassicMutationAccess
+    or
+    // class MyMutation < BaseMutation
+    exists(GraphqlRelayClassicMutationClass other |
+      other.getModule() = resolveScopeExpr(this.getSuperclassExpr())
+    )
+  }
+}
+
+/**
+ * A `ClassDeclaration` for a class that extends `GraphQL::Schema::Resolver`.
+ * For example,
+ *
+ * ```rb
+ * module Resolvers
+ *   class Base < GraphQL::Schema::Resolver
+ *     argument_class Arguments::Base
+ *   end
+ * end
+ *
+ * module Resolvers
+ *   class RecommendedItems < Resolvers::Base
+ *     type [Types::Item], null: false
+ *     argument :order_by, Types::ItemOrder, required: false
+ *
+ *     def resolve(order_by: )
+ *       # call your application logic here...
+ *     end
+ *   end
+ * end
+ * ```
+ */
+class GraphqlSchemaResolverClass extends ClassDeclaration {
+  GraphqlSchemaResolverClass() {
+    // class BaseResolver < GraphQL::Schema::Resolver
+    this.getSuperclassExpr() instanceof GraphqlSchemaResolverAccess
+    or
+    // class MyResolver < BaseResolver
+    exists(GraphqlSchemaResolverClass other |
+      other.getModule() = resolveScopeExpr(this.getSuperclassExpr())
+    )
+  }
+}
+
+class GraphqlSchemaObjectClass extends ClassDeclaration {
```

Comment:
As you mentioned already, having documentation for each of these classes would be great.